[TextareaAutosize] Typescript error TS2344

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:
Error occurs when building app using npm.
App uses React Typescript.


### Current behavior 😯

error TS2344 thrown:

node_modules/@mui/base/TextareaAutosize/TextareaAutosize.d.ts:14:93 - error TS2344: Type '"required" | "id" | "name" | "color" | "translate" | "value" | "hidden" | "form" | "slot" | "style" | "title" | "dir" | "accessKey" | "draggable" | "lang" | "className" | "prefix" | ... 253 more ... | "onResizeCapture"' does not satisfy the constraint 'keyof TextareaAutosizeProps'.
  Type '"onResize"' is not assignable to type 'keyof TextareaAutosizeProps'. Did you mean '"onReset"'?

### Expected behavior 🤔

_No response_

### Context 🔦

_No response_

### Your environment 🌎

System:
    OS: Windows 10 10.0.22621
  Binaries:
    Node: 16.17.0 - c:\program files\nodejs\node.EXE
    @mui/private-theming:  5.11.13
    @mui/styled-engine:  5.11.11
    @mui/system:  5.11.13
    @mui/types: ^7.2.3 => 7.2.3
    @mui/utils: ^5.11.13 => 5.11.13
    @types/react: ^17.0.33 => 17.0.52
    react: ^17.0.2 => 17.0.2
    react-dom: ^17.0.2 => 17.0.2
    typescript: ^4.4.4 => 4.9.3

my tsconfig
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "allowJs": false,
    "skipLibCheck": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["./src"]
}

--- a/packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx
+++ b/packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx
@@ -268,7 +268,7 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(
       />
     </React.Fragment>
   );
-});
+}) as React.ForwardRefExoticComponent<TextareaAutosizeProps & React.RefAttributes<Element>>;
 
 TextareaAutosize.propTypes /* remove-proptypes */ = {
   // ----------------------------- Warning --------------------------------

--- a/packages/mui-material-next/src/Button/TouchRipple.tsx
+++ b/packages/mui-material-next/src/Button/TouchRipple.tsx
@@ -275,7 +275,7 @@ const TouchRipple = React.forwardRef<TouchRippleActions, TouchRippleProps>(funct
       </TransitionGroup>
     </TouchRippleRoot>
   );
-});
+}) as React.ForwardRefExoticComponent<TouchRippleProps & React.RefAttributes<TouchRippleActions>>;
 
 TouchRipple.propTypes = {
   /**

